lib/components/url_view/url_view.dart

```diff
@@ -0,0 +1,149 @@
+      case TagFilterType.lastInboundTurn:
+        return 'last-inbound-turn-filter';
+    }
+    throw 'Trying to read an unknown filter type: $type';
+  }
+
+  String getQueryAfterDateFilterKey(TagFilterType type) {
+    switch (type) {
+      case TagFilterType.include:
+        return 'include-after-date';
+      case TagFilterType.exclude:
+        return 'exclude-after-date';
+      default:
+        throw 'Trying to read an unknown filter type: $type';
+    }
+  }
+
+  Set<String> getPageUrlFilterTags(TagFilterType type) {
+    var queryFilterKey = getQueryTagFilterKey(type);
+    var uri = Uri.parse(window.location.href);
+    if (uri.queryParameters.containsKey(queryFilterKey)) {
+      List<String> filterTags = uri.queryParameters[queryFilterKey].split(' ');
+      filterTags.removeWhere((tag) => tag == "");
+      return filterTags.toSet();
+    }
+    return Set();
+  }
+
+  void setPageUrlFilterTags(TagFilterType type, Set<String> filterTags) {
+    var queryFilterKey = getQueryTagFilterKey(type);
+    var uri = Uri.parse(window.location.href);
+    Map<String, String> queryParameters = new Map.from(uri.queryParameters);
+    if (filterTags == null || filterTags.isEmpty) {
+      queryParameters.remove(queryFilterKey);
+    } else {
+      queryParameters[queryFilterKey] = filterTags.join(' ');
+    }
+    uri = uri.replace(queryParameters: queryParameters);
+    window.history.pushState('', '', uri.toString());
+  }
+
+  String getPageUrlConversationList() {
+    var uri = Uri.parse(window.location.href);
+    if (uri.queryParameters.containsKey(queryConversationListKey)) {
+      return uri.queryParameters[queryConversationListKey];
+    }
+    return null;
+  }
+
+  void setPageUrlConversationList(String conversationListId) {
+    var uri = Uri.parse(window.location.href);
+    Map<String, String> queryParameters = new Map.from(uri.queryParameters);
+    if (conversationListId == null) {
+      queryParameters.remove(queryConversationListKey);
+    } else {
+      queryParameters[queryConversationListKey] = conversationListId;
+    }
+    uri = uri.replace(queryParameters: queryParameters);
+    window.history.pushState('', '', uri.toString());
+  }
+
+  String getPageUrlConversationId() {
+    var uri = Uri.parse(window.location.href);
+    if (uri.queryParameters.containsKey(queryConversationIdKey)) {
+      return uri.queryParameters[queryConversationIdKey];
+    }
+    return null;
+  }
+
+  void setPageUrlConversationId(String conversationId) {
+    var uri = Uri.parse(window.location.href);
+    Map<String, String> queryParameters = new Map.from(uri.queryParameters);
+    if (conversationId == null) {
+      queryParameters.remove(queryConversationIdKey);
+    } else {
+      queryParameters[queryConversationIdKey] = conversationId;
+    }
+    uri = uri.replace(queryParameters: queryParameters);
+    window.history.pushState('', '', uri.toString());
+  }
+
+  // TODO: More specific Function parameter
```

Comment:
@marianamarasoiu could you help me with this? The parameter onError is of type Function that takes Exception as input and returns nothing / void. I'm unable to come up with the right syntax.

Reply:
@elayabharath see below my suggestion, can you give it a try and let me know if it works?

Based on the way you were trying to specify the type, I think you're trying to use a syntax from somewhere else, maybe Javascript? in Dart, you always have to put the type before the name, and the value after it. 

`Function` is usually a type, but can also be an object (the object representing the type), which is why the way you wrote it didn't give you a syntax error, but trying to parametrise it would have done (because it won't be an object).

Hope this makes sense, but plmk if more explanation/examples would help!